server/src/main/java/org/apache/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -219,83 +218,148 @@ public void stop()
     }
   }
 
-  @Override
-  public boolean enableDataSource(final String dataSource)
+  private VersionedIntervalTimeline<String, DataSegment> getVersionedIntervalTimeline(final String dataSource, final Interval interval)
```

Comment:
I think `buildVersionedIntervalTimeline` would be a better name, with a method javadoc that explains that it builds a new versioned interval timeline from all segments (enabled or disabled) that overlap with the provided interval